goldmane/pkg/aggregator/ring_index.go

```diff
@@ -106,11 +115,15 @@ func (a *RingIndex) List(opts IndexFindOpts) []*types.Flow {
 		flows = flows[startIdx:endIdx]
 	}
 
-	return flows
+	return flows, totalPages
 }
 
 func (r *RingIndex) Add(d *types.DiachronicFlow) {
 }
 
 func (r *RingIndex) Remove(d *types.DiachronicFlow) {
 }
+
+func (a *RingIndex) Keys(opts IndexFindOpts) ([]int64, int) {
+	return []int64{}, 0
```

Comment:
Do we want a warning log here? Seems like a bug if we ever call this?

Reply:
I was just going off the precedence set above, but I guess this is different since you expect something valid to be returned.

I think a panic might actually be appropriate, if this is called then it's a dev error.

Reply:
Added a panic.